example/main.tf

```diff
@@ -0,0 +1,129 @@
+# =============================================
+#  Variables 
+# =============================================
+variable "env" {}
+
+variable "cluster_name" {
+  description = "Name for ECS Cluster"
+  type        = string
+  default     = "ecs-cluster"
+}
+
+variable "region" {
+  type = string
+}
+
+variable "tags" {
+  type        = map(string)
+  description = "common tags for all resources"
+  default     = {}
+}
+
+locals {
+  common_tags = merge(var.tags, {
+    env                 = var.env
+    terraform_managed   = "true"
+    terraform_workspace = terraform.workspace
+    ecs_cluster         = var.cluster_name
+  })
+}
+
+# =============================================
+#  Networking and resources
+# =============================================
+
+resource "tls_private_key" "ecs_root" {
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+module "vpc" {
```

Comment:
I think your example should probably assume a VPC exists rather than creating one. Creating a VPC is probably out of scope for this.

Reply:
mmmm I debated this and opted for an example that was easy to spin up, but I do like the idea of having the example being minimal

Reply:
Spinning up a new VPC is too much change for most environments to just test this out IMO.